Documentation/sources/content-card-ui/public-classes/uiaction.md

```diff
@@ -0,0 +1,103 @@
+# Sealed Class - UIAction
+
+Represents an action that can be performed on a UI component.
+
+#### Class Parameters
+
+| Parameter | Type                                         | Description                                                  |
+| --------- | -------------------------------------------- | ------------------------------------------------------------ |
+| T         | [AepUITemplate](./UIModels/aepuitemplate.md) | Represents a UI template model associated with the acction like SmallImageTemplate, which backs the composable on which the event has occurred. |
+| S         | [AepCardUIState](./State/aepcarduistate.md)  | Representing the state of the AEP card composable on which the event has occurred. |
+
+#### Public Properties
+
+| Property | Type                | Description                          |
+| -------- | ------------------- | ------------------------------------ |
+| aepUI    | [AepUI](./aepui.md) | The AepUI associated with the event. |
+
+#### Syntax
+
+``` kotlin
+sealed class UIEvent<T : AepUITemplate, S : AepCardUIState>(open val aepUi: AepUI<T, S>)
+```
+
+## Data Class - Display
+
+Event that represents the display of a UI element.
+
+#### Class Parameters
+
+| Parameter | Type                                         | Description                                                  |
+| --------- | -------------------------------------------- | ------------------------------------------------------------ |
+| T         | [AepUITemplate](./UIModels/aepuitemplate.md) | Represents a UI template model associated with the acction like SmallImageTemplate, which backs the composable on which the event has occurred. |
+| S         | [AepCardUIState](./State/aepcarduistate.md)  | Representing the state of the AEP card composable on which the event has occurred. |
```

Comment:
Since the class parameters need to be the same for all classes implementing `UIAction` , how do you feel about omitting it here and in `Interact` and `Dismiss`

Reply:
sounds good, will clean this up